docs/abi-spec.rst

```diff
@@ -233,7 +233,7 @@ Given the contract:
 ::
 
     pragma solidity >=0.4.16 <0.7.0;
-
+    // This will report a warning (pureness is not enforced yet on older versions of compiler)
```

Comment:
I will need to change this, as it will make cmd line tests fail (on latest compiler versions this will not cause a warning).

Is it k to make two sets of comments, one for latest compilers, one for old versions?

Reply:
I fear that these comments in the documentation might confuse people.

Is it possible to only check the warning condition on the latest compiler?

Reply:
It is possible to do that, yes. In that case, I would just need to ignore warnings when checking if examples are compiled correctly with minimal version

Reply:
Sounds good!